### PR TITLE
Merchant item update

### PR DESCRIPTION
**Pull request by samueldevine**
just one user story here!

- added edit_spec file to test the feature
- added 1 test in index_spec and fixed a bunch of broken ones
- added flash message displays to the view for all pages in `views/layouts/application.html.erb`

### CODE REVIEW

File: spec/features/merchant_items/edit_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    it 'after updating, I see a flash message indicating success' do
+      visit "/merchants/#{@merchant.id}/items/#{@item.id}/edit"
+
+      fill_in :name, with: 'tungsten cube'
+      fill_in :description, with: 'SUPER HEAVY'
+      fill_in :unit_price, with: 25000000
+      click_on 'Update Item'
+
+      expect(page).to have_content 'Item updated successfully'
```
Comment (haewonito) [inline spec/features/merchant_items/edit_spec.rb:47]: I am wondering what happens if the item won't be updated successfully. If the user puts in the wrong type of input for the attribute or something.  the user story doesn't mention it, so I guess we don't have to worry about it.  we didn't do validations for any 
 of the models, right?

File: app/controllers/merchant_items_controller.rb [hunk 2]
```diff
@@ -6,13 +6,19 @@ def index
...
 
   def update
-    @merchant = Merchant.find(params[:merchant_id])
+    merchant = Merchant.find(params[:merchant_id])
+    item = merchant.items.find(params[:item_id])
 
     if params[:change_item_status]
-      @item = @merchant.items.find(params[:item_id_for_change])
-      @item.change_status
+      item.change_status
```
Comment (haewonito) [inline app/controllers/merchant_items_controller.rb:13]: we could probably dry up the code here.  somehow merge change_item_status and other updates.  But seeing how I am still pretty lost with API, we should probably leave it alone.  lol.

File: spec/features/merchant_items/index_spec.rb [hunk 3]
```diff
@@ -4,37 +4,43 @@
   describe 'as a merchant' do
```
Comment (haewonito) [inline spec/features/merchant_items/index_spec.rb:4]: I couldn't follow everything that was happening here, to be honest. Too many changes. :) But thanks for refactoring and incorporating enums. Looks much neater.

File: app/controllers/merchant_items_controller.rb [hunk 4]
```diff
@@ -6,13 +6,19 @@ def index
...
+      redirect_to "/merchants/#{merchant.id}/items"
+    else
+      item.update(item_params)
+      redirect_to "/merchants/#{merchant.id}/items/#{item.id}"
+    end
 
-      redirect_to "/merchants/#{@merchant.id}/items"
+    if item.save
+      flash[:success] = 'Item updated successfully :)'
```
Comment (michab17) [inline app/controllers/merchant_items_controller.rb:21]: love the little smiley face

**Review by michab17 — APPROVED** [review]

**Pull request merged**